flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/inference/InputTypeStrategiesTest.java

```diff
@@ -481,7 +484,70 @@
 				"Map strategy fails for an odd number of arguments",
 				InputTypeStrategies.SPECIFIC_FOR_MAP)
 				.calledWithArgumentTypes(DataTypes.BIGINT(), DataTypes.BIGINT(), DataTypes.BIGINT())
-				.expectErrorMessage("Invalid number of arguments. 3 arguments passed.")
+				.expectErrorMessage("Invalid number of arguments. 3 arguments passed."),
+
+			TestSpec
+				.forStrategy(
+					"Logical type roots instead of concrete data types",
+					sequence(
+						logical(LogicalTypeRoot.VARCHAR),
+						logical(LogicalTypeRoot.DECIMAL),
+						logical(LogicalTypeRoot.DECIMAL),
+						logical(LogicalTypeRoot.BOOLEAN),
+						logical(LogicalTypeRoot.INTEGER, false)))
+				.calledWithArgumentTypes(
+					DataTypes.NULL(),
+					DataTypes.INT(),
+					DataTypes.DOUBLE(),
+					DataTypes.BOOLEAN().notNull(),
+					DataTypes.INT().notNull())
+				.expectSignature(
+					"f(<VARCHAR>, <DECIMAL>, <DECIMAL>, <BOOLEAN>, <INTEGER NOT NULL>)")
+				.expectArgumentTypes(
+					DataTypes.VARCHAR(1),
+					DataTypes.DECIMAL(10, 0),
+					DataTypes.DECIMAL(30, 15),
+					DataTypes.BOOLEAN(),
+					DataTypes.INT().notNull()),
+
+			TestSpec
+				.forStrategy(
+					"Logical type roots with wrong implicit cast",
+					sequence(logical(LogicalTypeRoot.VARCHAR)))
+				.calledWithArgumentTypes(DataTypes.INT())
+				.expectSignature("f(<VARCHAR>)")
+				.expectErrorMessage(
+					"Unsupported argument type. Expected type root 'VARCHAR' but actual type was 'INT'."),
+
+			TestSpec
+				.forStrategy(
+					"Logical type roots with wrong nullability",
+					sequence(logical(LogicalTypeRoot.VARCHAR, false)))
+				.calledWithArgumentTypes(DataTypes.VARCHAR(5))
+				.expectSignature("f(<VARCHAR NOT NULL>)")
+				.expectErrorMessage(
+					"Unsupported argument type. Expected nullable type of root 'VARCHAR' but actual type was 'VARCHAR(5)'."),
+
+			TestSpec
+				.forStrategy(
+					"Logical type family instead of concrete data types",
+					sequence(
+						logical(LogicalTypeFamily.CHARACTER_STRING),
+						logical(LogicalTypeFamily.EXACT_NUMERIC),
+						logical(LogicalTypeFamily.APPROXIMATE_NUMERIC),
```

Comment:
Can we add a case with e.g. `BIGINT` used with `APPROXIMATE_NUMERIC` family? This should work due to the implicit casts right?